src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageAction.java

```diff
@@ -151,9 +139,11 @@ protected ActionResult execute() throws EntityDoesNotExistException {
                 return createShowPageResult(
                         Const.ViewURIs.INSTRUCTOR_FEEDBACK_RESULTS_BY_GIVER_RECIPIENT_QUESTION, data);
             case "recipient-question-giver":
+                data.initForViewByRecipientQuestionGiver(instructor, data.selectedSection, data.showStats, data.groupByTeam);
                 return createShowPageResult(
                         Const.ViewURIs.INSTRUCTOR_FEEDBACK_RESULTS_BY_RECIPIENT_QUESTION_GIVER, data);
             case "giver-question-recipient":
+                data.initForViewByGiverQuestionRecipient(instructor, data.selectedSection, data.showStats, data.groupByTeam);
```

Comment:
Would it be better for the moment to place a call to a method in that *PageData that does the diverting for these init methods instead?
Also, is there a reason for passing in the three variables behind when they belong to the `data` object anyway? Preparation for setting them to private later?